lib/motions/general-motions.coffee

```diff
@@ -149,10 +149,16 @@ class MoveLeft extends Motion
 class MoveRight extends Motion
   operatesInclusively: false
 
-  moveCursor: (cursor, count=1) ->
+  # MoveRight can be combined with Delete, which will indicate it by the presence of options.deleting
+  moveCursor: (cursor, count=1, options={}) ->
     _.times count, =>
-      cursor.moveRight() unless cursor.isAtEndOfLine()
-      cursor.moveRight() if settings.wrapLeftRightMotion() and cursor.isAtEndOfLine()
+      goingToNextLine = atom.config.get('vim-mode.wrapLeftRightMotion')
+
+      if not cursor.isAtEndOfLine() # the line is not empty
+        cursor.moveRight()
+        goingToNextLine = false if options.deleting # when deleting, wrapLeftRightMotion only works on an empty line
```

Comment:
Just checked it and it's not simple:
 - `gUl` will not move the cursor in VIM, but in vim-mode it does (I may look into that)
 - `gUl` will do nothing in VIM on an empty line, in vim-mode with wrapping on it will move to the next line (and probably try to uppercase the newline which is a no-op)
 - `gU3l` in VIM will span a newline if done on an empty line or near the end of a line, so does vim-mode but it uppercases one more character than VIM does, in a way that's more consistent with the `l` motion so it doesn't bother me

 - `yl` in VIM on a non-empty line only yanks the character under the cursor, same in vim-mode
 - `yl` in VIM on an empty line yanks the newline and treats it as linewise for `p`; vim-mode does the same 
 - `y2l` in VIM on a non-empty line on the last character only yanks that character, not including the newline, in vim-mode it also yanks the newline which makes more sense to me (and `p` sees it as a linewise selection because that's what Utils.coffee does, can't blame it)
 - `y3l` in VIM on the last character will yank the character under the cursor, the newline, and the first character of the next line, and the same in vim-mode

None of the above is affected by this PR or should be. Does this make sense?